azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -699,6 +763,30 @@ func flattenEventGridEventSubscriptionStorageQueueEndpoint(input *eventgrid.Stor
 	return []interface{}{result}
 }
 
+func flattenEventGridEventSubscriptionAzureFunctionEndpoint(input *eventgrid.AzureFunctionEventSubscriptionDestination) []interface{} {
+	results := make([]interface{}, 0)
+
+	if input == nil {
+		return results
+	}
+
+	result := make(map[string]interface{})
+
+	if input.ResourceID != nil {
+		result["id"] = *input.ResourceID
+	}
+
+	if input.MaxEventsPerBatch != nil {
+		result["max_events_per_batch"] = *input.MaxEventsPerBatch
+	}
+
+	if input.PreferredBatchSizeInKilobytes != nil {
+		result["preferred_batch_size_in_kilobytes"] = *input.PreferredBatchSizeInKilobytes
+	}
+
+	return append(results, result)
```

Comment:
we need to ensure these fields always have a value in the state - as such could we make this:

```suggestion
	functionAppId := ""
	if input.ResourceID != nil {
		functionAppId = *input.ResourceID
	}
	
	maxEventsPerBatch := 0
	if input.MaxEventsPerBatch != nil {
 		maxEventsPerBatch = *input.MaxEventsPerBatch
 	}
 	
	prefferedBatchSize := 0
	if input.PreferredBatchSizeInKilobytes != nil {
 		prefferedBatchSize = *input.PreferredBatchSizeInKilobytes
 	}
	
	return append(results, map[string]interface{}{
		"function_app_id": functionAppId,
		"max_events_per_batch": maxEventsPerBatch,
		"preferred_batch_size_in_kilobytes": preferredBatchSize,
	})
```